arangod/Replication2/Supervision/CollectionGroupSupervision.cpp

```diff
@@ -529,6 +538,9 @@ struct TransactionBuilder {
                   "/arango/Plan/CollectionGroups/", database, "/", gid.id(),
                   "/collections/", action.cid))
               .inc("/arango/Plan/Version")
+              .precs()
+              .isNotEmpty(basics::StringUtils::concatT(
+                  "/arango/Target/CollectionGroups/", database, "/", gid.id()))
```

Comment:
For the `DropCollectionPlan` action, I think this precondition might not be the best fit.

Reply:
I find it an odd choice, since by now I'm so used to the fact the once something is "gone" from `Target`, we remove it from `Plan`. I would've expected the precondition to be more like `isEmpty`, because we want to remove the collection from `Plan` only if it is no longer in `Target`.
This is more of a question than a strong opinion, as I'm not very familiar with the supervision behind collection groups.
Other than that, I think the PR looks good!

Reply:
I understand. If we decide to drop a collection from plan, it is already gone from Target. It is undefined behavior if we add a collection with the same ID to the target again. We might not notice it, we might have already deleted the data.

Reply:
_If we decide to drop a collection from plan, it is already gone from Target._ - I agree.
But, the precondition:
```
.isNotEmpty(basics::StringUtils::concatT(
                  "/arango/Target/CollectionGroups/", database, "/", gid.id()))
```
Doesn't this translate to: "Drop the collection from plan, only when the Target entry is not empty"?